invenio_search_ui/views.py

```diff
@@ -44,9 +44,72 @@ def sorted_options(sort_options):
     ]
 
 
+def searchkit_sort_options(sort_options, default_sort):
+    """Sort and map sort options for display.
+
+        :param sort_options: A dictionary containing the field name as key and
+            asc/desc as value.
+        :returns: A dictionary with sorting options for Invenio-Search-JS
+        with react-searchkit.
+        """
+    return [
+        {
+            'text': v['title'],
+            'sortBy': k,
+            'sortOrder': v.get('default_order', 'asc'),
+            'default': default_sort["query"] == k,
+            'defaultOnEmptyString': default_sort["noquery"] == k,
+        }
+        for k, v in
+        sorted(sort_options.items(), key=lambda x: x[1].get('order', 0))
+    ]
+
+
+def searchkit_aggs(aggs):
+    """Format the aggregation configuration to be used in Invenio-Search-JS.
+
+    :param aggs: A dictionary with aggregation configuration
+    :returns: A formated dictionary.
+    """
+    return [
+        {
+            'title': k.capitalize(),
+            'aggName': k,
+            'field': v['terms']['field'],
+        }
+        for k, v in aggs.items()
+    ]
+
+
 @blueprint.app_template_filter('format_sortoptions')
 def format_sortoptions(sort_options):
     """Create sort options JSON dump for Invenio-Search-JS."""
     return json.dumps({
         'options': sorted_options(sort_options)
     })
+
+
+@blueprint.app_template_filter('format_config')
+def format_config(config, endpoint_name):
+    """Create config JSON dump for Invenio-Search-JS."""
+
+    # TODO check what happens when search_index not defined
+    # in record_rest_endpoints
+    search_index = config.get('RECORDS_REST_ENDPOINTS', {}).get(
+        endpoint_name, {}).get('search_index', 'records')
+
+    config_sort_options = config.get('RECORDS_REST_SORT_OPTIONS', {}).get(
+        search_index)
+    config_default_sort = config.get('RECORDS_REST_DEFAULT_SORT', {}).get(
+        search_index)
+
+    return json.dumps({
```

Comment:
Minor: Use the Jinja filter ``|tojson`` instead. It's not a big issue since there's no user-provided data, but ``tojson`` does proper escaping to prevent XSS attacks, so that the JSON can safely be included in HTML script tags. 

I can see this is an inherited problem from the module :-)

Reply:
fixed